component/azstorage/utils.go

```diff
@@ -309,6 +309,36 @@ func parseMetadata(attr *internal.ObjAttr, metadata map[string]*string) {
 	}
 }
 
+func AddMetadata(metadata Metadata, key, value string) bool {
+	if _, ok := metadata[key]; ok {
+		return false
+	}
+
+	if key != "" {
+		capitalizedKey := strings.ToUpper(string(key[0])) + key[1:]
```

Comment:
General convention to convert case is to always convert it to lower and not upper.